### PR TITLE
Use system dependent line separators.

### LINKED ISSUE #580: Use OS-dependent line ending for generated messages
Per convention, a well-behaved Java program should generate text in the native format for the platform it runs on, i.e. lines ended by \r\n on Windows, \r on Mac, \n on others. JCommander does not do that when generating error/usage messages, but has \n hard-soldered in. (14 occurrences in `DefaultUsageFormatter`, `JCommander` and `UnixStyleUsageFormatter`.)

The system dependent line separator is  available as `System.getProperty("line.separator")`. I suggest this to be used instead of "\n". On request, I will be happy to submit a PR.

### PR DESCRIPTION
**Pull request by bengtmartensson**
Resolves #580.

### CODE REVIEW

**Review by mkarg — CHANGES_REQUESTED** [review]
Thank you for your contribution! ❤️ 

In fact, your code change is **very** intrusive, and I need to say, it *unnecessarily* complex for such a simple problem and makes the code *unnecessarily* harder to read. In fact, it also voids separation of concerns, i. e. it consideres system specifics all over the *source* places, which makes not much sense, as we could simply re-encode EOL at the few *target* locations where actually *output*, like in the form of a system-dependent output stream wrapper.

Having said that, I would kindly ask you to overhaul your solution to *not* modify the string creation everywhere, but solely replace EOL where actually rendered on a system-dependend output.

**Comment by mkarg** [comment]
Could you please provide a unit test that proofs that the *final* outcome (when rendered on STDOUT) is actually wrong? I assume you noticed it on Windows, so it would be good to have a unit test that currently fails on Windows.

File: src/main/java/com/beust/jcommander/DefaultUsageFormatter.java [hunk 1]
```diff
@@ -67,7 +67,7 @@ public final void usage(String commandName, StringBuilder out, String indent) {
 
         if (description != null) {
             out.append(indent).append(description);
-            out.append("\n");
+            out.append(System.getProperty("line.separator"));
```
Comment (yeikel) [inline src/main/java/com/beust/jcommander/DefaultUsageFormatter.java:70]: `System.lineSeparator()` is a bit better than this

Comment (mkarg) [reply src/main/java/com/beust/jcommander/DefaultUsageFormatter.java:70]: ...and still as unnecessary, as I already explained. ;-)

Can you actually reproduce the reported problem? On which OS? In which shell?

**Comment by bengtmartensson** [comment]
Closing, see https://github.com/cbeust/jcommander/issues/580#issuecomment-2094366630

File: src/main/java/com/beust/jcommander/DefaultUsageFormatter.java (hunk 1 above)
Comment (bengtmartensson) [reply src/main/java/com/beust/jcommander/DefaultUsageFormatter.java:70]: @yeikel : `System.lineSeparator()` always returns the same `String`, while `System.getProperty("line.separator")` can be changed by `System.setProperty("line.separator")`, which allows you to test emulating a foreign system. (so I do not have to fire up my windows computer .)

**Pull request closed without merging**